src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java

```diff
@@ -314,6 +312,40 @@ else if (exercise instanceof ModelingExercise) {
         return studentExams;
     }
 
+    /**
+     * Helper method to return a map for each user to their exercises
+     *
+     * @param studentExams the student exams of the users containing the exercises
+     * @return a map of the User as key, and a list of the users exercises as value
+     */
+    public Map<User, List<Exercise>> getExercisesOfUserMap(Set<StudentExam> studentExams) {
+        return studentExams.stream()
+                .collect(
+                        Collectors
+                                .toMap(StudentExam::getUser,
+                                        studentExam -> studentExam.getExercises().stream().filter(exercise -> exercise instanceof ModelingExercise
+                                                || exercise instanceof TextExercise || exercise instanceof FileUploadExercise || exercise instanceof ProgrammingExercise)
```

Comment:
Did you consider instead of using multiple `instanceof`s using a single negated one? E.g. `!(exercise instanceof QuizExercise)`? This makes the condition exclusive instead of inclusive, where I am not sure if we want this, but performancewise a single instanceof is better than multiple.

Reply:
Performance wise you are correct probably, although I don't know how much difference it would make, probably not a lot. For now I think it's better to keep the condition inclusive. Also, when we would add new exercise types this might be a possible point of failure.